### ISSUE TITLE
Build failure with GHC 8.4

### ISSUE THREAD

**Issue by DanBurton** [timbod7/haskell-chart#181]
Add semigroup instances.
See also: https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#SemigroupMonoidsuperclasses

```
Building library for Chart-1.8.3..
[ 1 of 37] Compiling Graphics.Rendering.Chart.Geometry ( Graphics/Rendering/Chart/Geometry.hs, dist/build/Graphics/Rendering/Chart/Geometry.o )

Graphics/Rendering/Chart/Geometry.hs:162:6: error:
    • No instance for (Semigroup Path) arising from a use of ‘<>’
    • In the expression:
        moveTo p1 <> lineTo p2 <> lineTo p3 <> lineTo p4 <> close
      In the expression:
        let
          p2 = Point x1 y2
          p4 = Point x2 y1
        in moveTo p1 <> lineTo p2 <> lineTo p3 <> lineTo p4 <> close
      In an equation for ‘rectPath’:
          rectPath (Rect p1@(Point x1 y1) p3@(Point x2 y2))
            = let
                p2 = Point x1 y2
                p4 = Point x2 y1
              in moveTo p1 <> lineTo p2 <> lineTo p3 <> lineTo p4 <> close
    |
162 |   in moveTo p1 <> lineTo p2 <> lineTo p3 <> lineTo p4 <> close
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Graphics/Rendering/Chart/Geometry.hs:199:10: error:
    • No instance for (Semigroup Path)
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Monoid Path’
    |
199 | instance Monoid Path where
    |          ^^^^^^^^^^^
```

**Comment by DanBurton** [comment]
Ping @tfausak, note that this build failure keeps Chart out of the Stackage nightly build plan.

**Comment by barryfm** [comment]
I've made a similar set of patches in my fork.  I have added backend patches creating chart-gi-cairo, and chart-gi-gtk, which depend on haskell-gi rather than gtk2hs.  However, it uses for gi-cairo the haskell-gi/cairo interface (which has not been released) rather that the haskell-gi/bindings/cairo that is in hackage.

The chart-gi-* backends do not work with ghc-8.2.x, since I used  haskell-gi-overloading which is disabled in the ghc-8.2.x releases.  I did build and use the chart-gi-cairo and chart-gi-gtk with ghc-8.4.2 without warning and with the existing gtk2hs backends with Ubuntu 18.04 (ghc-8.0.2) without warning, but with some warnings about redundant declarations with Debian Stretch (ghc-8.0.1).

I tried building, as the haskell-gi wiki suggested, using the the gtk2hs bindings on top of the gi-cairo
interface, but had a lot of problems, so I did not complete the port.  It seems too much effort for an interm solution.

The gtk2hs developers seem to suggest using the haskell-gi interface in all new software projects,
and at some point the haskell-gi project intends to come up with a solution like their haskell-gi/cabal
code, although there seems to be no release date.

With the ghc developers intention of having new releases every 6 months, it seems this state of API changes and broken code is with us for the indefinite future.
